src/GovUk.Education.ExploreEducationStatistics.Data.Processor/Functions/PoisonQueueHandler.cs

```diff
@@ -0,0 +1,52 @@
+using System;
+using System.Collections.Generic;
+using GovUk.Education.ExploreEducationStatistics.Data.Importer.Services.Interfaces;
+using GovUk.Education.ExploreEducationStatistics.Data.Model.Database;
+using GovUk.Education.ExploreEducationStatistics.Data.Processor.Model;
+using GovUk.Education.ExploreEducationStatistics.Data.Processor.Services.Interfaces;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Extensions.Logging;
+
+namespace GovUk.Education.ExploreEducationStatistics.Data.Processor.Functions
+{
+    public class PoisonQueueHandler
+    {
+        private readonly IBatchService _batchService;
+
+        public PoisonQueueHandler(IBatchService batchService)
+        {
+        }
+
+        [FunctionName("ProcessUploadsPoisonHandler")]
+        public void ProcessUploadsPoisonQueueHandler(
+            [QueueTrigger("imports-pending-poison")]
+            ImportMessage message,
+            ILogger logger
+        )
+        {
+            var errors = new List<string>() { "File failed to import for unknown reason in upload processing stage."};
+            
+            _batchService.FailImport(
+                message.Release.Id.ToString(),
+                message.DataFileName,
+                errors
+            ).Wait();
```

Comment:
I think if the ProcessUploadsPoisonQueueHandler is made async then the Wait can be removed & use await instead which I've read is non-blocking unlike wait I think.

Reply:
It can yes, done.